client/resource_attribute_restriction.go

```diff
@@ -0,0 +1,413 @@
+package anaml
+
+import (
+	"errors"
+	"strconv"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+const attributeDescription = `# Attribute Restrictions
+
+An Attribute is a key/value pair for user-defined metadata. Restrictions limit the attributes
+that can be applied to a given object, and what values they can take.
+
+Multiple different types of attributes are supported:
+
+- Enum ("Choice")
+- Free Text
+- Boolean
+- Integer
+`
+
+func ResourceAttributeRestriction() *schema.Resource {
+	return &schema.Resource{
+		Description: attributeDescription,
+		Create:      resourceAttributeRestrictionCreate,
+		Read:        resourceAttributeRestrictionRead,
+		Update:      resourceAttributeRestrictionUpdate,
+		Delete:      resourceAttributeRestrictionDelete,
+		Importer:    &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"key": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringIsNotWhiteSpace,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"enum": {
+				Type:         schema.TypeList,
+				Optional:     true,
+				MaxItems:     1,
+				Elem:         enumAttributeSchema(),
+				ExactlyOneOf: []string{"enum", "freetext", "boolean", "integer"},
+			},
+			"freetext": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem:     &schema.Resource{},
+			},
+			"boolean": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem:     &schema.Resource{},
+			},
+			"integer": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem:     &schema.Resource{},
+			},
+			"applies_to": {
+				Type:     schema.TypeList,
```

Comment:
This should be of type Set, as the order should not matter.

Reply:
the same should be true of the enum choices then, i think?

Reply:
Yeah.